core/variant/array.cpp

```diff
@@ -49,6 +49,10 @@ class ArrayPrivate {
 	ContainerTypeValidate typed;
 };
 
+// Array to use in optional parameters on methods.
+// Ex: DEFVAL(Array::default_array)
+Array Array::default_array = Array::create_read_only();
```

Comment:
I'd suggest declaring this instead in a file belonging to the binding system, renamed to `default_out_array_arg`, so instead of having the somewhat blurry concept of a default array, we'd have the concept of array-for-default-arguments-in-bindings.

Then, there could be a macro like this, that can be used in bindings to make the syntax a bit less jarring:
`#define DEF_OUT_ARRAY DEFVAL(default_out_array_arg)`

The names are just a suggestion to illustrate my point. I'm not advocating for them as they are exactly written here.

Reply:
I moved the default array in ClassBD. I'm not that familiar with the binding system but it's seems a good place.

I create the `DEFVAL_ARRAY` macro. I did not used `OUT` in the name because it could be used as an input default parameter also.

Also, putting the code in `ClassDB` made a lot more sense to create the function `ClassDB::is_default_array_arg` which could be used to verify if an array is the default value.

Let me know what you think about these changes.

Reply:
Looks great!